cpp/include/cugraph/edge_properties.hpp

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```

Comment:
And I think we should re-consider this file name.

We already have `edge_property.hpp` and this file name can be confusing.

And `numeric_(arithmetic_)device_uvector_t` may be used in other places than just edge properties. Should we better rename this file (something like arithmetic_variant_types.hpp)?